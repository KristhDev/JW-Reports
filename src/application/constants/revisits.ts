export const revisitsMessages = {
    ABOUT_MIN_LENGTH: 'La información de la persona debe tener al menos 10 caracteres.',
    ABOUT_REQUIRED: 'La información de la persona es requerida.',
    ADDED_SUCCESS: 'Ha agregado la revisita correctamente.',
    ADDRESS_MIN_LENGTH: 'La dirección debe tener al menos 10 caracteres.',
    ADDRESS_REQUIRED: 'La dirección es requerida.',
    COMPLETED_SUCCESS: 'Ha marcado como completa su revisita correctamente.',
    DELETED_SUCCESS: 'Ha eliminado la revisita correctamente.',
    EXPORTED_SUCCESS: 'Ha exportado sus revisitas correctamente. El archivo se encuentra en la carpeta de descargas de su dispositivo.',
    NEXT_VISIT_REQUIRED: 'La fecha de la último visita no puede estar vacía.',
    PERSON_MIN_LENGTH: 'El nombre de la persona debe tener al menos 2 caracteres.',
    PERSON_NAME_REQUIRED: 'El nombre de la persona es requerido.',
    UNSELECTED_COMPLETE: 'No hay una revisita seleccionada para completar.',
    UNSELECTED_DELETE: 'No hay una revisita seleccionada para eliminar.',
    UNSELECTED_UPDATE: 'No hay una revisita seleccionada para actualizar.',
    UPDATED_SUCCESS: 'Ha actualizado la revisita correctamente.',
}