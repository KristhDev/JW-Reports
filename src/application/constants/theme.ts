import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];