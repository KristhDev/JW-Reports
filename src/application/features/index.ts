export * from './auth';
export * from './courses';
export * from './lessons';
export * from './permissions';
export * from './preaching';
export * from './revisits';
export * from './status';
export * from './types';
export * from './ui';