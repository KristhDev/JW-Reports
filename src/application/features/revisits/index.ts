export {
    default as revisitsReducer,
    addRevisit,
    addRevisits,
    clearRevisits,
    INIT_REVISIT,
    removeRevisit,
    removeRevisits,
    REVISITS_INITIAL_STATE,
    setHasMoreRevisits,
    setIsLastRevisitLoading,
    setIsRevisitDeleting,
    setIsRevisitsExporting,
    setIsRevisitLoading,
    setIsRevisitsLoading,
    setLastRevisit,
    setRefreshRevisits,
    setRevisitFilter,
    setRevisits,
    setRevisitsPagination,
    setRevisitsScreenHistory,
    setSelectedRevisit,
    updateRevisit
} from './slice';

export * from './types';