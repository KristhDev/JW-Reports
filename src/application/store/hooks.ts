import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();