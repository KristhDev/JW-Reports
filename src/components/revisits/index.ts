export { RevisitCard } from './RevisitCard';
export { RevisitForm } from './RevisitForm';