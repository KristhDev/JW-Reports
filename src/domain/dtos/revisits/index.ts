export * from './complete.dto';
export * from './create.dto';
export * from './update.dto';