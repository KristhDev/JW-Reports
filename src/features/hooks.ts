import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;