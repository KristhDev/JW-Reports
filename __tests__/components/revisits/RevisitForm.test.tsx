import React from 'react';
import { Image } from 'react-native-image-crop-picker';
import { act, fireEvent, render, screen, waitFor } from '@testing-library/react-native';

/* Features */
import { revisitsState, selectedRevisitState  } from '../../features/revisits';

/* Components */
import { RevisitForm } from '../../../src/components/revisits';

/* Hooks */
import { useImage, useRevisits, useStatus, useTheme } from '../../../src/hooks';

/* Theme */
import { darkColors } from '../../../src/theme';

const setErrorFormMock = jest.fn();
const saveRevisitMock = jest.fn();
const updateRevisitMock = jest.fn();
const takeImageToGalleryMock = jest.fn();
const takePhotoMock = jest.fn();

const image: Image = {
    height: 200,
    mime: 'image/jpeg',
    path: 'https://local-image.com/images.jpg',
    size: 120,
    width: 200
}

const personName = 'Clifton DAmore';
const about = 'Possimus magnam cum quo saepe et accusamus consectetur molestiae. Eos et et nobis dolor. Enim repellat quia officia fuga qui. Cumque delectus unde possimus consequatur ducimus.';
const direction = 'Et fuga eaque voluptatum est. Et ad quos sit id ducimus a hic. Perferendis deserunt eius aut harum nisi doloremque consequuntur maiores. Aut voluptas corrupti facere sed ut. Enim qui quaerat quos quia quia expedita et.';

/* Mock hooks */
jest.mock('../../../src/hooks/useImage.ts');
jest.mock('../../../src/hooks/useRevisits.ts');
jest.mock('../../../src/hooks/useStatus.ts');
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <RevisitForm /> component', () => {
    (useImage as jest.Mock).mockReturnValue({
        image,
        takeImageToGallery: takeImageToGalleryMock,
        takePhoto: takePhotoMock
    });

    (useRevisits as jest.Mock).mockReturnValue({
        state: {
            ...revisitsState,
            selectedRevisit: {
                ...revisitsState.selectedRevisit,
                next_visit: '2022-12-29 00:00:00'
            }
        },
        saveRevisit: saveRevisitMock,
        updateRevisit: updateRevisitMock
    });

    (useStatus as jest.Mock).mockReturnValue({
        setErrorForm: setErrorFormMock
    });

    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors },
        BUTTON_TRANSPARENT_COLOR: 'rgba(255, 255, 255, 0.15)'
    });

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('should to match snapshot', async () => {
        await waitFor(() => {
            render(<RevisitForm />);
        });

        await act(() => {
            expect(screen.toJSON()).toMatchSnapshot();
        });
    });

    it('should call setErrorForm when form is invalid', async () => {
        await waitFor(() => {
            render(<RevisitForm />);
        });

        await act(async () => {
            await waitFor(() => {

                /* Get submit touchable */
                const touchable = screen.getAllByTestId('button-touchable')[3];
                fireEvent.press(touchable);

                /* Check if setErrorForm is called one time */
                expect(setErrorFormMock).toHaveBeenCalledTimes(1);
            });
        });
    });

    it('should call saveRevisit when form is valid and selectedRevisit is empty', async () => {
        await waitFor(() => {
            render(<RevisitForm />);
        });

        await act(async () => {
            await waitFor(() => {

                /* Get text inputs to type person name, about and direction */
                const inputs = screen.getAllByTestId('form-field-text-input');
                fireEvent(inputs[0], 'onChangeText', personName);
                fireEvent(inputs[1], 'onChangeText', about);
                fireEvent(inputs[2], 'onChangeText', direction);

                /* Get submit touchable */
                const touchable = screen.getAllByTestId('button-touchable')[3];
                fireEvent.press(touchable);

                /* Check if setErrorForm is called one time */
                expect(saveRevisitMock).toHaveBeenCalledTimes(1);
            });

            /* Get text of submit touchable */
            const btnText = screen.getAllByTestId('button-text')[3];

            /**
             * Check if text of submit touchable is equal to Guardar and saveRevisit
             * is called with respective args
             */
            expect(btnText.props.children).toBe('Guardar');
            expect(saveRevisitMock).toHaveBeenCalledWith({
                revisitValues: {
                    person_name: personName,
                    about,
                    address: direction,
                    next_visit: expect.any(Date)
                },
                image
            });
        });
    });

    it('should call updateRevisit when form is valid and selectedRevisit isnt empty', async () => {

        /* Mock data of useRevisits */
        (useRevisits as jest.Mock).mockReturnValue({
            state: selectedRevisitState,
            saveRevisit: saveRevisitMock,
            updateRevisit: updateRevisitMock
        });

        await waitFor(() => {
            render(<RevisitForm />);
        });

        await act(async () => {
            await waitFor(() => {

                /* Get text input to type new content of about */
                const inputs = screen.getAllByTestId('form-field-text-input');
                fireEvent(inputs[1], 'onChangeText', about);

                /* Get submit touchable */
                const touchable = screen.getAllByTestId('button-touchable')[3];
                fireEvent.press(touchable);

                /* Check if updateRevisit is called one time */
                expect(updateRevisitMock).toHaveBeenCalledTimes(1);
            });

            /* Get text of submit touchable */
            const btnText = screen.getAllByTestId('button-text')[3];

            /**
             * Check if text of submit touchable is equal to Actualizar and
             * updateRevisit is called with respective args
             */
            expect(btnText.props.children).toBe('Actualizar');
            expect(updateRevisitMock).toHaveBeenCalledWith(
                {
                    person_name: selectedRevisitState.selectedRevisit.person_name,
                    about,
                    address: selectedRevisitState.selectedRevisit.address,
                    next_visit: expect.any(Date)
                },
                image
            );
        });
    });

    it('should call takeImageToGallery when gallery button is pressed', async () => {
        await waitFor(() => {
            render(<RevisitForm />);
        });

        await act(async () => {
            await waitFor(() => {

                /* Get gallery touchable */
                const touchable = screen.getAllByTestId('button-touchable')[0];
                fireEvent.press(touchable);

                /* Check if takeImageToGallery is called one time */
                expect(takeImageToGalleryMock).toHaveBeenCalledTimes(1);
            });
        });
    });

    it('should call takePhoto when photo button is pressed', async () => {
        await waitFor(() => {
            render(<RevisitForm />);
        });

        await act(async () => {
            await waitFor(() => {

                /* Get camera touchable */
                const touchable = screen.getAllByTestId('button-touchable')[1];
                fireEvent.press(touchable);

                /* Check if takePhoto is called one time */
                expect(takePhotoMock).toHaveBeenCalledTimes(1);
            });
        });
    });
});