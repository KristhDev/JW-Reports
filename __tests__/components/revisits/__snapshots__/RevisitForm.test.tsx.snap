// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <RevisitForm /> component should to match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingBottom": 40,
      "paddingTop": 30,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "padding": 10,
        "width": 675,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "marginBottom": 5,
            "padding": 5,
          },
          undefined,
        ]
      }
      testID="form-field-label"
    >
      Nombre de la persona:
    </Text>
    <View
      style={
        Object {
          "borderColor": "transparent",
          "borderRadius": 7,
          "borderWidth": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "#FFFFFF",
            "borderRadius": 5,
            "borderWidth": 0.5,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "transparent",
                "borderRadius": 5,
                "borderWidth": 1.5,
                "flexDirection": "row",
                "paddingRight": 10,
              },
              undefined,
            ]
          }
        >
          <TextInput
            autoCorrect={false}
            cursorColor="#C0A7E1"
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSelectionChange={[Function]}
            placeholder="Ingrese el nombre"
            placeholderTextColor="#BFBFBF"
            selection={
              Object {
                "end": 0,
                "start": 0,
              }
            }
            selectionColor="#A0B9E2"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "flex": 1,
                  "fontSize": 17,
                  "paddingLeft": 10,
                  "paddingRight": 5,
                  "textAlignVertical": "center",
                },
                undefined,
              ]
            }
            testID="form-field-text-input"
            value=""
          />
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#BFBFBF",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "padding": 10,
        "width": 675,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "marginBottom": 5,
            "padding": 5,
          },
          undefined,
        ]
      }
      testID="form-field-label"
    >
      Información de la persona:
    </Text>
    <View
      style={
        Object {
          "borderColor": "transparent",
          "borderRadius": 7,
          "borderWidth": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "#FFFFFF",
            "borderRadius": 5,
            "borderWidth": 0.5,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "transparent",
                "borderRadius": 5,
                "borderWidth": 1.5,
                "flexDirection": "row",
                "paddingRight": 10,
              },
              undefined,
            ]
          }
        >
          <TextInput
            autoCorrect={false}
            cursorColor="#C0A7E1"
            multiline={true}
            numberOfLines={10}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSelectionChange={[Function]}
            placeholder="Ingrese datos sobre la persona, tema de conversación, aspectos importantes, etc..."
            placeholderTextColor="#BFBFBF"
            selection={
              Object {
                "end": 0,
                "start": 0,
              }
            }
            selectionColor="#A0B9E2"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "flex": 1,
                  "fontSize": 17,
                  "paddingLeft": 10,
                  "paddingRight": 5,
                  "textAlignVertical": "top",
                },
                undefined,
              ]
            }
            testID="form-field-text-input"
            value=""
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "padding": 10,
        "width": 675,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "marginBottom": 5,
            "padding": 5,
          },
          undefined,
        ]
      }
      testID="form-field-label"
    >
      Dirección:
    </Text>
    <View
      style={
        Object {
          "borderColor": "transparent",
          "borderRadius": 7,
          "borderWidth": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "#FFFFFF",
            "borderRadius": 5,
            "borderWidth": 0.5,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "transparent",
                "borderRadius": 5,
                "borderWidth": 1.5,
                "flexDirection": "row",
                "paddingRight": 10,
              },
              undefined,
            ]
          }
        >
          <TextInput
            autoCorrect={false}
            cursorColor="#C0A7E1"
            multiline={true}
            numberOfLines={4}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSelectionChange={[Function]}
            placeholder="Ingrese la dirección"
            placeholderTextColor="#BFBFBF"
            selection={
              Object {
                "end": 0,
                "start": 0,
              }
            }
            selectionColor="#A0B9E2"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "flex": 1,
                  "fontSize": 17,
                  "paddingLeft": 10,
                  "paddingRight": 5,
                  "textAlignVertical": "top",
                },
                undefined,
              ]
            }
            testID="form-field-text-input"
            value=""
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "padding": 10,
        "width": 675,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 18,
          "marginBottom": 5,
          "padding": 5,
        }
      }
    >
      Foto
    </Text>
    <Image
      source={
        Object {
          "uri": "https://local-image.com/images.jpg",
        }
      }
      style={
        Object {
          "borderRadius": 5,
          "height": 750,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginTop": 20,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            false,
            Array [
              Object {
                "backgroundColor": "#C0A7E1",
                "borderRadius": 10,
              },
              undefined,
            ],
          ]
        }
        testID="button-touchable"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 20,
                "paddingVertical": 8,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                Object {
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "color": "#121212",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 2,
                    "textAlign": "center",
                  },
                  undefined,
                ],
              ]
            }
            testID="button-text"
          >
            Galería
          </Text>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#121212",
                  "fontSize": 25,
                },
                Object {
                  "marginLeft": 5,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            false,
            Array [
              Object {
                "backgroundColor": "#C0A7E1",
                "borderRadius": 10,
              },
              undefined,
            ],
          ]
        }
        testID="button-touchable"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 20,
                "paddingVertical": 8,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                Object {
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "color": "#121212",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 2,
                    "textAlign": "center",
                  },
                  undefined,
                ],
              ]
            }
            testID="button-text"
          >
            Cámara
          </Text>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#121212",
                  "fontSize": 25,
                },
                Object {
                  "marginLeft": 5,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "padding": 10,
        "width": 675,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "marginBottom": 5,
            "padding": 5,
          },
          undefined,
        ]
      }
      testID="datetimefield-label"
    >
      Próxima visita:
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#FFFFFF",
              "borderRadius": 5,
              "borderWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
            },
            undefined,
          ]
        }
      >
        <TextInput
          autoCorrect={false}
          editable={false}
          modalTitle="Próxima visita"
          placeholder="Seleccione el día"
          placeholderTextColor="#BFBFBF"
          selectionColor="#A0B9E2"
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 17,
                "paddingLeft": 10,
              },
              undefined,
            ]
          }
          testID="datetimefield-text-input"
          value="29/12/2022"
        />
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            false,
            Array [
              Object {
                "backgroundColor": "#C0A7E1",
                "borderRadius": 10,
              },
              Object {
                "marginLeft": 10,
              },
            ],
          ]
        }
        testID="button-touchable"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 20,
                "paddingVertical": 8,
              },
              Object {
                "paddingHorizontal": 10,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                Object {
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "color": "#121212",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 2,
                    "textAlign": "center",
                  },
                  undefined,
                ],
              ]
            }
            testID="button-text"
          />
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#121212",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      isVisible={false}
      onHide={[Function]}
      transparent={true}
      visible={false}
    />
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        false,
        Array [
          Object {
            "backgroundColor": "#C0A7E1",
            "borderRadius": 10,
          },
          Object {
            "marginTop": 30,
          },
        ],
      ]
    }
    testID="button-touchable"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 20,
            "paddingVertical": 8,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            Object {
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "#121212",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 2,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
        testID="button-text"
      >
        Guardar
      </Text>
    </View>
  </View>
</View>
`;
